docs/Design/Alpha/Data Model.md

```diff
@@ -0,0 +1,215 @@
+        string added_at "Timestamp"
+        string updated_at "Timestamp"
+    } 
+    user {
+        int id PK 
+        string(31) username UK ""
+        string(255) email UK ""
+        string(31) phone "Nullable"
+        string created_at "Timestamp"
+        string updated_at "Timestamp"
+    }
+    user_purchase_album {
+        int id PK
+        int user_id FK
+        int album_id FK
+        int purchase_price "Nullable"
+        string(255) invoice_uri UK
+        string purchased_at "Timestamp"
+    }
+    user_purchase_track {
+        int id PK
+        int user_id FK
+        int track_id FK
+        int purchase_price "Nullable"
+        string(255) invoice_uri UK
+        string purchased_at "Timestamp"
+    }
+    tag {
+        int id PK
+        string(31) slug UK "Unique human-readable URL-safe identifier"
+        string(31) name "The pretty name of the Tag"
+    }
+    link {
+        int id PK 
+        string(255) url
+        int artist_id FK
+    }
+    
+    artist }o--|{ user : "artist_manager"
+    artist }|--o{ album : "artist_album"
+    artist }o--o{ tag : "tag_artist"
+    artist ||--o{ link : "" 
+    artist }o--o{ artist : "artist_members"
+    album }|--|{ track : "album_track"
+    album }o--o{ tag : "tag_album"
+    track }|--o{ artist : "track_artist"
+    track }o--o{ tag : "tag_track"
+    playlist ||--o{ playlist_track : ""
+    playlist_track }o--|| track : ""
+    user }o--o{ artist : "user_following"
+    user }o--o{ album : "user_collection_album"
+    user }o--o{ track : "user_collection_track"
+    user ||--o{ playlist : ""
+    user_purchase_album }o--|| album : ""
+    user_purchase_album }o--|| user : ""
+    user_purchase_track }o--|| track : ""
+    user_purchase_track }o--|| user : ""
+```
+
+## Notes
+
+- While most of the many-to-many relationship tables are minimally simple (two columns, each containing IDs),
+  there are a few (`playlist_track`, `user_purchase_album`, `user_purchase_track`) that are
+  defined explicitly within the diagram because they are a touch more complex.
+
+### Artists
+- If an **Artist** is a group, it can be linked to individual **Artist**s as members so that **User**s
+can browse the **Artist**'s solo work or work with other groups. **User**s will also be able to browse 
+the groups of which an **Artist** is a member from the individual **Artist**'s page.
+- An **Artist** may be without a manager if a **User** uploads an **Album** and credits an artist who
+doesn't have an account. A dummy **Artist** entity would be created and would somehow have to be claimed
+by the real artist in the event they do decide to make an account.
+
+### Album
+- Allows **Artist**s to specify a `release_year` if the exact `release_date` is unknown
+
+### Tracks
+- The price for a **Track** is nullable in case the **Artist** doesn't want to allow individual **Track**s
+to be purchased. If the **Artist** wants the **Track** to be free, the price can be set to zero.
```

Comment:
How would we support "pay what you want" pricing with this model? We can add that as a feature to any album or track with a free price point but it should be noted that other platforms also support "pay what you want" with a minimum as well. We may need a new field in the ER diagram to compensate for this

Reply:
I think a flag should be simple enough.